src/utils/useStickyMargin.ts

```diff
@@ -0,0 +1,106 @@
+import { useEffect } from "react";
+
+/* Code taken from https://www.tpgi.com/prevent-focused-elements-from-being-obscured-by-sticky-headers/.
+Requires data attribute (data-sticky-header) on header, and could be refactored to work for sticky footer as well.  
+applyStickyMargin() checks whether a focused element needs a sticky margin by calculating boundary interactions, then 
+applies that margin (pushing page up) accordingly. applyStickyMargin() is called by a focus event on any page element or a 
+ResizeObserver event on the header. */
+
+export function useStickyMargin() {
+  useEffect(() => {
+    const applyStickyMargin = (entries, observer, sticky) => {
+      const focused = (document.activeElement || document.body) as HTMLElement;
+
+      let applicable = focused !== document.body;
+      if (applicable && sticky.header) {
+        applicable = !sticky.header.contains(focused);
+      }
+      if (applicable && sticky.selector) {
+        applicable = focused.matches(sticky.selector);
+      }
+
+      if (applicable) {
+        const edge = {
+          header: sticky.header
+            ? sticky.header.getBoundingClientRect().bottom + sticky.offset!
```

Comment:
Is the exclamation point at the end a typo?

Reply:
Yes :(